src/proto/h1/role.rs

```diff
@@ -190,6 +191,7 @@ impl Http1Transaction for Server {
         let mut is_te = false;
         let mut is_te_chunked = false;
         let mut wants_upgrade = subject.0 == Method::CONNECT;
+        let mut header_case_map = HeaderCaseMap::default();
```

Comment:
Do you know if creating this even if not needed has any affect on server peformance? Might be easiest to test that with `cargo bench --bench pipeline` (that's the easiest, though I find it's slightly noisier than actually starting up a pipeline server and running `wrk` at it, but that's also more tedious...).

Reply:
I didnt bother checking this because that's just like what `Client::parse` does :)

Reply:
Yea, fair. Though also, (while I grumble about it all the time) the server benchmarks are watched and generally used as PR for the project, so it'd probably be good to be aware, at least.

Reply:
I've put both `HeaderCaseMap::default` calls in their own `if ctx.preserve_header_case` block just to be sure.